src/main/java/seedu/address/model/person/Experience.java

```diff
@@ -40,7 +40,11 @@ public Experience(String experience) {
      * Returns true if a given number matches validation.
      */
     public static boolean isValidExperience(String test) {
-        return test.matches(VALIDATION_REGEX);
+        if (test.matches(VALIDATION_REGEX)) {
+            int value = Integer.parseInt(test);
+            return (value <= MAX_EXPERIENCE);
+        }
+        return false;
```

Comment:
What happens if the user inputs a long value that cannot be parsed into an integer, for example `"1111111111111111"`? I believe this will cause a `NumberFormatException` that isn't accounted for. For mine, I added a try-catch block to catch any `NumberFormatException`, and returns false if it happens.

Also, I think you should add a test case of inputting a long value (which should assertFalse) into `ExperienceTest.java` as well!

Reply:
Agree, will handle that exception and add test cases :)